Sources/Support/StoreMessageType.swift

```diff
@@ -28,29 +29,35 @@ import StoreKit
     var numberValue: NSNumber {
         return NSNumber(value: self.rawValue)
     }
+
 }
 
 #if os(iOS) || targetEnvironment(macCatalyst) || VISION_OS
 
-@available(iOS 16.4, *)
+@available(iOS 16.0, *)
 @available(macOS, unavailable)
 @available(watchOS, unavailable)
 @available(tvOS, unavailable)
 extension Message.Reason {
+
     var messageType: StoreMessageType? {
         switch self {
-        // billingIssue message reason was added in iOS 16.4, but it's not recognized by older xcode versions.
-        // https://developer.apple.com/documentation/xcode-release-notes/xcode-14_3-release-notes
-        #if swift(>=5.8)
-        case .billingIssue: return .billingIssue
-        #endif
         case .priceIncreaseConsent: return .priceIncreaseConsent
         case .generic: return .generic
         default:
+            // billingIssue message reason was added in iOS 16.4, but it's not recognized by older xcode versions.
+            // https://developer.apple.com/documentation/xcode-release-notes/xcode-14_3-release-notes
+            #if swift(>=5.8)
+            if #available(iOS 16.4, *), case .billingIssue = self {
+                return .billingIssue
+            }
+            #endif
```

Comment:
I refactored this to allow using 16.0 everywhere @tonidero 
This is unnecessarily complex because `Message.Reason` isn't an `enum`.

Reply:
clever approach